File: app-java/src/test/java/tech/thdev/java_udemy_sample/_01_constructor/_03_ConstructorInit.java
package tech.thdev.java_udemy_sample._01_constructor;

/**
 * Created by tae-hwan on 09/10/2016.
 *
 * 생성자에서의 초기화
 */

public class _03_ConstructorInit {

    public static class InnerClass {

        private String name;

        public InnerClass() {
            name = "InnerClass";
        }

        public String getName() {
            return name;
        }
    }
}

File: app-kotlin/src/test/java/tech/thdev/kotlin_udemy_sample/class_01/_03_ConstructorInit.kt
package tech.thdev.kotlin_udemy_sample.class_01

/**
 * Created by tae-hwan on 09/10/2016.
 *
 * 생성자에서의 초기화
 */

class _03_ConstructorInit {

    class InitClass() {
//        var name: String = "InitClass"
        var name: String

        // 생성자인 constructor에서는 초기화를 하지 못하고, init에서는 가능
        init {
            name = "InitClass"
        }

        // 아래와 같이 초기화 하여야 name의 값이 변경 됨
        constructor(name: String) : this() {
            this.name = name
        }
    }

    class ConstructorInit(val name: String) {
        // ...
    }
}